src/main/java/edu/iis/powp/adapter/SpecialLinePlotterAdapter.java

```diff
@@ -0,0 +1,49 @@
+package edu.iis.powp.adapter;
+
+import edu.iis.client.plottermagic.AbstractPlotter;
+import edu.iis.client.plottermagic.IPlotter;
+import edu.iis.powp.appext.ApplicationWithDrawer;
+import edu.kis.powp.drawer.shape.ILine;
+import edu.kis.powp.drawer.shape.LineFactory;
+
+import java.util.logging.Logger;
+
+
+public class SpecialLinePlotterAdapter extends AbstractPlotter implements IPlotter
+{
+    private ILine line = LineFactory.getBasicLine();
+    private final static Logger LOGGER = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+
+    public SpecialLinePlotterAdapter(int x, int y) {
+        super(x,y);
+    }
+
+    @Override
+    public void drawTo(int x, int y) {
+    	line.setStartCoordinates(this.getX(), this.getY());
+        line.setEndCoordinates(x, y);
+        this.setPosition(x,y);
+
+        ApplicationWithDrawer.getDrawPanelController().drawLine(line);
```

Comment:
1. Constructor injection should be used (using a parameter). This is unnecessary direct coupling with ApplicationWithDrawer class.